### DISCUSSION TITLE
disjoint graphs in dataloader

### DISCUSSION THREAD

**Discussion by Shiiivaa** [pyg-team/pytorch_geometric#5127] [Q&A] [answered]
Hello,
Let's say I have generated 500 graphs with the same structure (adj. matrix is the same). Now, I concatenate all these graphs into one giant Graph in a list:
`Graph =[graph1, graph2, graph3, ... , graph500]`

To start the training process, I need to batch my data, so I use Dataloader:
`loader = DataLoader(dataset = Graph , batch_size=10, shuffle=True)`

Here are my questions:
1. Is this the right way to concatenate the graphs so they would not have any connections with each other? I need to have disjoint graphs in each batch.
2. Once we set shuffle as True in Dataloader, does it mean that in each epoch, Dataloader shuffles the batcehs? If yes, Is there a recommended method of obtaining the indices of the graphs that Dataloder sample? I need to see which graphs (out of 500 graphs) are sampled by Dataloader.

Thank you in advance for your help.

**Comment by Padarn** [comment, marked as answer]
> Is this the right way to concatenate the graphs so they would not have any connections with each other? I need to have disjoint graphs in each batch.

Yes this will batch them together so they are disjoint

> Once we set shuffle as True in Dataloader, does it mean that in each epoch, Dataloader shuffles the batcehs? If yes, Is there a recommended method of obtaining the indices of the graphs that Dataloder sample? I need to see which graphs (out of 500 graphs) are sampled by Dataloader.

The batch index is not maintained as far as I know (I could be wrong). You could make the graph index a property of the graph, or add it to `y`.
